tools/istio-iptables/pkg/cmd/run.go

```diff
@@ -432,9 +413,10 @@ func (iptConfigurator *IptablesConfigurator) run() {
 		// Avoid infinite loops. Don't redirect Envoy traffic directly back to
 		// Envoy for non-loopback traffic.
 		iptConfigurator.iptables.AppendRuleV4(constants.ISTIOOUTPUT, constants.NAT, "-m", "owner", "--uid-owner", uid, "-j", constants.RETURN)
-	}
+	} else {
+		// TPROXY uses GID
```

Comment:
I am not sure this is safe, we may use GID in iptables mode as well. Also, why do we not need `split` anymore, I think it can be a list?
cc @JimmyCYJ

Reply:
I had added GID matching in the first place because it was required for `TPROXY`. It's not used for `REDIRECT` AFAIK.

Reply:
I think https://www.hyrumslaw.com/ may be in play, I am fairly sure @JimmyCYJ is depend on this. Which its fine to say "don't do that" and I agree, but we should be explicit about doing it with release note, etc